packages/overlay/src/integrations/sentry/components/insights/agents/AITraceList.tsx

```diff
@@ -0,0 +1,124 @@
+import { useMemo } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import { ReactComponent as Sort } from "~/assets/sort.svg";
+import { ReactComponent as SortDown } from "~/assets/sortDown.svg";
+import type { SpotlightAITrace } from "~/integrations/sentry/types";
+import classNames from "~/lib/classNames";
+import Table from "~/ui/table";
+import { AGENT_HEADERS, AGENT_SORT_KEYS } from "../../../constants";
+import { SearchProvider } from "../../../context/SearchContext";
+import { useSpotlightAITraces } from "../../../data/useSpotlightAITraces";
+import useSort from "../../../hooks/useSort";
+import useSentryStore from "../../../store";
+import AITraceDetail from "./AITraceDetails";
+import AITraceItem from "./AITraceItem";
+
+type AgentSortTypes = (typeof AGENT_SORT_KEYS)[keyof typeof AGENT_SORT_KEYS];
+type AgentComparator = (a: SpotlightAITrace, b: SpotlightAITrace) => number;
+
+const COMPARATORS: Record<AgentSortTypes, AgentComparator> = {
+  [AGENT_SORT_KEYS.timestamp]: (a, b) => a.timestamp - b.timestamp,
+  [AGENT_SORT_KEYS.duration]: (a, b) => a.durationMs - b.durationMs,
+  [AGENT_SORT_KEYS.traceId]: (a, b) => a.id.localeCompare(b.id),
```

Comment:
Don't think we should be doing `localCompare` for ids. I expect it to be much slower.